File: 4-FrogRiverOneOptime.js
function solution(X, A) {
  const occupiedPositions = new Map();
  let totalSum = (X * (X + 1)) / 2;
 console.log(occupiedPositions.has(A[0]))
  for (let i = 0; i < A.length; i++) {
    if (!occupiedPositions.has(A[i]) && A[i] <= X) {
      occupiedPositions.set(A[i], true);
      totalSum -= A[i];
      console.log(totalSum)
      if (totalSum === 0) {
        return i;
      }
    }
  }
  return -1;
}



console.log(solution(5,[1,3,6,1,4,2,3,5,4,6,7,8]))

File: 4-FrogRiverOneOptime.py

from pickle import TRUE


def solution(X, A):
    occupedPosition=dict()
    totalSum=(X*(X+1))/2
    
    for i in range(len(A)):
        if occupedPosition.get(A[i])== None and A[i] <= X:
            occupedPosition[A[i]]=True
            totalSum -= A[i]
            print(totalSum)
            if totalSum == 0:
                return i



    return -1
    pass







print(solution(5,[1,3,6,1,4,2,3,5,4,6,7,8]))
